### DISCUSSION TITLE
DCE Demo Discussions: `stampino-element` by Justin Fagnani

### DISCUSSION THREAD

**Discussion by Westbrook** [w3c/webcomponents-cg#90] [Ideas]
As presented on 25 March to the WCCG, `stampino-element` is a proposed approach to declarative custom elements by @justinfagnani.

https://docs.google.com/presentation/d/13mIqxQJVlQZxBrtYAsTP-WjhSbFcWO5p4a-NCDFBdhc/edit?usp=sharing 

Further discussion and questions below.

**Comment by jogibear9988** [comment]
Why do you not add the style to the template tag? Is it so it only needs to be parsed once?

**Reply by justinfagnani** [reply]
So that it becomes a constructible stylesheet and is actually shared among the instances. Ultimately I want these declarative style modules to be able to be shared across component definitions.

**Comment by jogibear9988** [comment]
And why do you not use "eval" for evaluating the expressions?

**Reply by justinfagnani** [reply]
I don't think we want to allow just any JS in expressions because then we don't have a reduction in power and complexity that many people want with declarative elements.

A simpler expression language could be implemented in non-JS environments, so we can have non-JS SSR. And if sandboxed enough to prevent gadget attacks, these elements could run with script disabled.

**Reply by jogibear9988** [reply]
I thought this also earlier, but then we wanted to do more and more in the Binding expressions. And in the old polymer days, it was really painfull that you had to create a computedProperty for every thing wich was more than a simple access

**Comment by jogibear9988** [comment]
Do you also support upgradeing the elemnt/Template etc at runtime (with a new instanciation of the defining element), or do you think this is bad? (I also think normally it should not be needed, but for use in my designer it is usefull)

For example, see your demo in my designer: [stampino demo in designer](https://node-projects.github.io/web-component-designer-demo/index.html?npm=stampino-element&html=%3Cstampino-element%20name=%22simple-greeter%22%20properties=%22name%22%3E%20%3Cstyle%20type=%22adopted-css%22%3E%20:host%20{%20color:%20blue;%20}%20%3C/style%3E%20%3Ctemplate%3E%20%3Ch1%3EHello%20{{%20name%20}}!%3C/h1%3E%20%3C/template%3E%20%3C/stampino-element%3E%20%3Csimple-greeter%20name=%22World%22%3E%3C/simple-greeter%3E)
If you change the template after loading the element nothing will happen.
Here is the question, should happen something?
